packages/vscode/src/extension.ts

```diff
@@ -33,14 +33,23 @@ export async function activate(context: vscode.ExtensionContext) {
 
       const currentWorkspacePath = normalizeFsPath(currentWorkspaceUri.fsPath);
 
-      const branchNames = await getBranchNames(gitPath, currentWorkspacePath);
-      const initialBaseBranchName = getBaseBranchName(branchNames);
-
       const githubToken: string | undefined = await getGithubToken(secrets);
       if (!githubToken) {
         throw new GithubTokenUndefinedError("Cannot find your GitHub token. Retrying github authentication...");
       }
 
+      const fetchBranchList = async () => {
+        const storedBranchList = context.workspaceState.get<string[]>("branchList") ?? [];
+        context.workspaceState.update(
+          "branchList",
+          (await getBranchNames(gitPath, currentWorkspacePath)) ?? storedBranchList
+        );
```

Comment:
여기 로직 관련해서 질문이 있는데요~
caching되어 있는 branch 관련 데이터가 없다면 getBranchNames 함수를 호출하는 방식으로 해야 하지 않을까요??
```suggestion
const branchListData = context.workspaceState.get<string[]>("branchList") ?? (await getBranchNames(gitPath, currentWorkspacePath));
context.workspaceState.upsate("branchList", branchListData);
```
이런 식으로요!

Reply:
네네 맞습니다! 짜다보니 caching이 아니라 오류 대체 fallback store 처럼 되어버리긴 했습니다. 🥲

전 일단 기존에 저장된 부분을 previous, 새로 fetching하는 부분을 latest라고 생각했습니다.
현재는 refresh랑 기본 fetching이랑 구분이 없다보니..branch list의 상태가 바뀌는 것을 업데이트 하기 위해 매번 fetching 하는 방식 + fetching에 혹시 오류가 있을 경우에는 기존 저장된 previous 값이 있다면 대체하는 방향으로 로직을 작성하였습니다.

어제 잠시 이야기 나눈 대로, refresh랑 fetch part 로직을 분리하는게 좋겠네요! 일단은 단순 fetch 함수에는 caching을 적용해 놓겠습니다 :) refresh part 작성 나눠서 같이 진행해봐요~

Reply:
cache를 하려면, 기존 데이터가 바뀌지 않았다는 전제가 깔려있어야 하는데,
아직 그걸(브랜치 상태) 모니터링하는 부분이 없어서, 조금 아리까리하네요.
(실제 브랜치 리스트의 변경이 있어도 계속 처음 caching된 데이터만 가져올테니)

일단 브랜치 리스트 가져오는 것이 cost가 큰 작업이 아닌 것 같으니, cache는 추후에 고민해봐도 좋을 것 같습니다!

Reply:
@ss-won 
네네! refresh, fetch 로직 분리하는 PR 얼른 올려보도록 할게용